elasticsearch.go

```diff
@@ -19,21 +19,38 @@ package elasticsearch
 
 import (
 	"encoding/base64"
+	"encoding/json"
 	"errors"
 	"fmt"
 	"net/http"
 	"net/url"
 	"os"
+	"regexp"
+	"strconv"
 	"strings"
+	"sync"
 	"time"
 
 	"github.com/elastic/go-elasticsearch/v7/esapi"
 	"github.com/elastic/go-elasticsearch/v7/estransport"
 	"github.com/elastic/go-elasticsearch/v7/internal/version"
 )
 
+var (
+	reVersion          *regexp.Regexp
+	useHeaderCheckOnly bool
+	productChecked     bool
+)
+
+func init() {
+	versionPattern := `(\d+).(\d+).(\d+)`
```

Comment:
Do we need a `^` to start this pattern? Also `.` should be escaped so it's not used as any character.

Does Go do special things with unicode when using `\d` like Python does? If so `[0-9]` may be preferable.